ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -73,7 +82,10 @@ public class DiskManager {
   private final MessageStoreHardDelete hardDelete;
   private final List<String> unexpectedDirs = new ArrayList<>();
   private final AccountService accountService;
+  private final DiskManagerConfig diskManagerConfig;
   private boolean running = false;
+  private DiskHealthStatus diskHealthStatus = DiskHealthStatus.UNKNOWN_HEALTH;
```

Comment:
why init to unknown ? init to GOOD. The healthcheck will set it to BAD

Reply:
Because if the ambry healthcheck endpoint was accessed and the disk healthcheck job hasn't finished running the first time then it would appear as though all the disks are healthy when they aren't. We need confirmation that the disk healthcheck passed before we can say the disk is healthy and return a GOOD response to ambry healthcheck